app/components/Modal/BorrowModal.jsx

```diff
@@ -227,6 +227,35 @@ class BorrowModalContent extends React.Component {
         );
     }
 
+    _maximizeDebt() {
+        let backing_balance = !this.props.backing_balance
+            ? {balance: 0, id: null}
+            : this.props.backing_balance.toJS();
+        let backingBalance = backing_balance.id
+            ? ChainStore.getObject(backing_balance.id)
+            : null;
+        window.s = backingBalance;
+        let backingAmount = backingBalance ? backingBalance.get("balance") : 0;
+        const maximumCollateral =
```

Comment:
You're still not taking into account any existing collateral here. The max collateral should be `currentCollateral + availableBalance - 10`.

Reply:
Right, I keep forgetting that you can have pre-existing collateral (from already borrowing an asset).  Will fix!